File: data-structures/python/src/static_array.py
class StaticArray:
    def __init__(self, size, default=None):
        if not isinstance(size, int) or size < 0:
            raise ValueError("size must be a non-negative integer")
        self._size = size
        self._data = [default] * size

    def at(self, index):
        if index < 0 or index >= self._size:
            raise IndexError("StaticArray.at: index out of range")
        return self._data[index]

    def set_at(self, index, value):
        if index < 0 or index >= self._size:
            raise IndexError("StaticArray.at: index out of range")
        self._data[index] = value

    def __getitem__(self, index):
        return self._data[index]

    def __setitem__(self, index, value):
        self._data[index] = value

    def front(self):
        return self._data[0]

    def back(self):
        return self._data[-1]

    def data(self):
        if self._size == 0:
            return None
        return self._data

    def size(self):
        return self._size

    def empty(self):
        return self._size == 0

    def fill(self, value):
        for i in range(self._size):
            self._data[i] = value

    def __iter__(self):
        return iter(self._data)

    def __len__(self):
        return self._size

File: data-structures/go/src/staticarray/static_array.go
package staticarray

import "errors"

var ErrOutOfRange = errors.New("StaticArray: index out of range")

type StaticArray[T any] struct {
	data []T
	size int
}

func New[T any](size int) *StaticArray[T] {
	return &StaticArray[T]{
		data: make([]T, size),
		size: size,
	}
}

func (a *StaticArray[T]) At(index int) (T, error) {
	var zero T
	if index < 0 || index >= a.size {
		return zero, ErrOutOfRange
	}
	return a.data[index], nil
}

func (a *StaticArray[T]) SetAt(index int, value T) error {
	if index < 0 || index >= a.size {
		return ErrOutOfRange
	}
	a.data[index] = value
	return nil
}

func (a *StaticArray[T]) Get(index int) T {
	return a.data[index]
}

func (a *StaticArray[T]) Set(index int, value T) {
	a.data[index] = value
}

func (a *StaticArray[T]) Front() T {
	return a.data[0]
}

func (a *StaticArray[T]) Back() T {
	return a.data[a.size-1]
}

func (a *StaticArray[T]) Data() []T {
	if a.size == 0 {
		return nil
	}
	return a.data
}

func (a *StaticArray[T]) Size() int {
	return a.size
}

func (a *StaticArray[T]) IsEmpty() bool {
	return a.size == 0
}

func (a *StaticArray[T]) Fill(value T) {
	for i := range a.data {
		a.data[i] = value
	}
}
